packages/contracts-bedrock/test/setup/ForkLive.s.sol

```diff
@@ -49,20 +49,33 @@ contract ForkLive is Deployer {
 
     /// @notice Forks, upgrades and tests a production network.
     /// @dev This function sets up the system to test by:
-    ///      1. reading the superchain-registry to get the contract addresses we wish to test from that network.
-    ///      2. deploying the updated OPCM and implementations of the contracts.
-    ///      3. upgrading the system using the OPCM.upgrade() function.
+    ///      0. reading the environment variable to determine the base and OP chain names, read the STATE_PATH data
+    /// created from superchain ops.
+    ///      1. if the environment variable is set, load the state from the given path.
+    ///      2. read the superchain-registry to get the contract addresses we wish to test from that network.
+    ///      3. deploying the updated OPCM and implementations of the contracts.
+    ///      4. upgrading the system using the OPCM.upgrade() function if not using the applied state from superchain
+    /// ops.
     function run() public {
-        // Read the superchain registry and save the addresses to the Artifacts contract.
-        _readSuperchainRegistry();
-
-        // Now deploy the updated OPCM and implementations of the contracts
-        _deployNewImplementations();
+        bool useOpsRepo = bytes(vm.envOr("STATE_PATH", string(""))).length > 0;
+        if (useOpsRepo) {
+            console.log("ForkLive: loading state from %s", vm.envString("STATE_PATH"));
+            vm.loadAllocs(vm.envString("STATE_PATH"));
```

Comment:
I think it would be useful to include a comment here to say why the order of commands matters here e.g. why don't we 'loadAllocs' _after_ '_readSuperchainRegistry'?

Reply:
Right, I believe @blmalone is suggesting to add that as a code comment which is a good idea :) 

Also one clarification would be "...from the newly updated values instead of reading the current implementation addresses from the forked chain", since the implementations are read via calls here, not read from the superchain registry